ObservableList script

SAMPROO opened the pull request.

SAMPROO committed:
ObservableList script

oelnenimmat commented on MorkoUnityProject/Assets/Scripts/ObservableList.cs:
```diff
@@ -0,0 +1,37 @@
+﻿using System;
```
Add header, aka own name and purpose of file/class in an comment at the top of file like
/*
Coder Champion

This is a class for containing abandoned souls
*/

oelnenimmat commented on MorkoUnityProject/Assets/Scripts/ObservableList.cs:
```diff
@@ -0,0 +1,37 @@
+    public int Count()
+    {
+        return this._data.Count;
+    }
```
Change this to a property, as is customary in c#. It can just directly return _data.Count
```suggestion
    public int Count => _data.Count;
```

oelnenimmat reviewed (changes requested):
Readonly on _data was good idea.

SAMPROO committed:
PR Changes

- Class description
- ObservableList() and indexer changed to lambda function
- Count function changed to a variable that returns List<>.Count

oelnenimmat commented on MorkoUnityProject/Assets/Scripts/ObservableList.cs:
```diff
@@ -1,22 +1,21 @@
+This class crates a generic list with the functionality of adding, removing at index and returning the size
+When Add() or RemoveAt(), Event is triggered
+*/
+
```
Move this at the very top, before any code(including using-statements. Also Add your name here, So we can easily know who to ask in case of questions.

oelnenimmat commented on MorkoUnityProject/Assets/Scripts/ObservableList.cs:
```diff
@@ -29,9 +28,6 @@ public void RemoveAt(int index, bool fromEnd = false)
\ No newline at end of file
+
+    public int Count => this._data.Count;
+}
```
By convention, variables and properties go before functions, move this above add and remove.

oelnenimmat reviewed (changes requested):
Good work, little more to go.

Side note: ObservableList(), indexer and Count in this form are not lamdas, they are called 'expression body', or expression bodied functions or properties. And Count is in fact a property, not a variable. I mentioned these in case you want to dig deeper, so now you have the right keywords to go :)

SAMPROO committed:
PR Tweaks

- Moved class description to the top and added contact
- Moved expression bodied property Count above Add() and RemoveAt()

SAMPROO committed:
Fullname
